### DISCUSSION TITLE
how to force fetching of channel fee information updates?

### DISCUSSION THREAD

**Discussion by AndySchroder** [lightningnetwork/lnd#7947] [Troubleshooting]
I have 3 nodes:

1. A has one channel to B and is not connected at the peer to peer level to C
2. B has one channel to both A and C (it is the routing node)
3. C has one channel to B

On node B I updated the fees on channel AB and channel BC. Node A saw the fee update right away for channel BC. Node C did not see the fee update for channel AB. I am testing this by running `lncli getchaninfo`. I have waited several hours. I've also sent a payment from A to C and from C to A.

Questions

1. Is there  command in LND that I can use to force fetching of new fee information from a node that it is interested in routing a payment through? Seems like this would be possible for a node that your are directly connected to, and it seems that if you are deliberately running `lncli getchaninfo` it should do so every time.
2. Why does routing a payment through a node not cause LND to force updating of fee information?
3. Why does node B not broadcast the fee updates on its own channels to all directly connected peers?
4. Is it possible to make a request for fee information through onion routed messages, or only with direct peers?
5. Do we have to wait for peers to gossip the information and there is no way to request an update? How long does this wait normally take?

**Comment by AndySchroder** [comment]
The updated fee information did eventually propagate after 24 hours, but that is way too long to wait.

**Reply by guggero** [reply]
What versions of `lnd` are those nodes? There were some fixes to the gossip relay/batching done recently.
There currently is no protocol message defined in the BOLT for _fetching_ graph related data. Those are only sent to the network through gossip. So if your node doesn't "hear" a message for some reason, it needs to wait for the next time that message is broadcast.
How long such a re-broadcast takes depends on the message type (I think). 

Another reason for not getting messages in time could be that you are running into a throttling mechanism. Are you sending out a lot (e.g. triggered by a script or tool) of channel fee updates? Meaning that you change your channels' fees very often? Then it could be that the messages from that node are throttled by your peers.
